app/mailers/notification_mailer.rb

```diff
@@ -6,7 +6,10 @@ class NotificationMailer < ApplicationMailer
          :routing
 
   before_action :process_params
-  before_action :set_status, only: [:mention, :favourite, :reblog]
+  with_options only: %i(mention favourite reblog) do
+    before_action :set_status
+    after_action :thread_by_conversation!
```

Comment:
This kind of looks like it wouldn't work, as it effectively moves the call after `mail` calls… but it seems we have tests for that behavior and they pass, so I'm not really against it, but it does feel a bit surprising to me.

Reply:
Yes - there are separate callback groups for the "building the mail" and "delivering the mail" groups ... the after_action is basically the same as the before action, but you have access to all the ivars or anything else done in the main body (and I guess if you had any final post-mail-building but pre-delivery stuff).